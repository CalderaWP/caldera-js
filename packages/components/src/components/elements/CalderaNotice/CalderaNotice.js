import React from 'react';

import { Notice } from '@wordpress/components';
export const CalderaNotice = ({ children, isError }) => (
	<Notice status={isError ? 'error' : 'success'}>{children}</Notice>
);
