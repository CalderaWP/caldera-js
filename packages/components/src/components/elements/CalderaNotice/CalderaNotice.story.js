import React from 'react';
import { storiesOf } from '@storybook/react';
import { CalderaNotice } from './CalderaNotice';

storiesOf('CalderaNotice', module).add('Success message', () => (
	<CalderaNotice isError={false}>Good Things!</CalderaNotice>
));

storiesOf('CalderaNotice', module).add('Error message', () => (
	<CalderaNotice isError={true}>Bad Things!</CalderaNotice>
));
