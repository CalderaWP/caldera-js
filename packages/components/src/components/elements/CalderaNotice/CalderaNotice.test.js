import * as React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { CalderaNotice } from './CalderaNotice';

describe('CalderaNotice', () => {
	it('Matches snapshot with Success message', () => {
		const component = renderer.create(
			<CalderaNotice isError={false}>Good Things!</CalderaNotice>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Matches snapshot with error message', () => {
		const component = renderer.create(
			<CalderaNotice isError={true}>Bad Things!</CalderaNotice>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
});
