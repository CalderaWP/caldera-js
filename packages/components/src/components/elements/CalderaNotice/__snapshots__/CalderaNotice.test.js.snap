// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalderaNotice Matches snapshot with Success message 1`] = `
<div
  className="components-notice is-success is-dismissible"
>
  <div
    className="components-notice__content"
  >
    Good Things!
  </div>
  <button
    aria-label="Dismiss this notice"
    className="components-button components-icon-button components-notice__dismiss"
    onClick={[Function]}
    type="button"
  >
    <svg
      aria-hidden="true"
      className="dashicon dashicons-no"
      focusable="false"
      height={20}
      role="img"
      viewBox="0 0 20 20"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12.12 10l3.53 3.53-2.12 2.12L10 12.12l-3.54 3.54-2.12-2.12L7.88 10 4.34 6.46l2.12-2.12L10 7.88l3.54-3.53 2.12 2.12z"
      />
    </svg>
  </button>
</div>
`;

exports[`CalderaNotice Matches snapshot with error message 1`] = `
<div
  className="components-notice is-error is-dismissible"
>
  <div
    className="components-notice__content"
  >
    Bad Things!
  </div>
  <button
    aria-label="Dismiss this notice"
    className="components-button components-icon-button components-notice__dismiss"
    onClick={[Function]}
    type="button"
  >
    <svg
      aria-hidden="true"
      className="dashicon dashicons-no"
      focusable="false"
      height={20}
      role="img"
      viewBox="0 0 20 20"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12.12 10l3.53 3.53-2.12 2.12L10 12.12l-3.54 3.54-2.12-2.12L7.88 10 4.34 6.46l2.12-2.12L10 7.88l3.54-3.53 2.12 2.12z"
      />
    </svg>
  </button>
</div>
`;
